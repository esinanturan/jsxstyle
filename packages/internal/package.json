{
  "name": "@jsxstyle/internal",
  "private": true,
  "version": "0.0.0",
  "main": "src/index.ts",
  "type": "module",
  "dependencies": {
    "@types/invariant": "^2.2.37",
    "invariant": "^2.2.4",
    "superstruct": "^2.0.2",
    "zx": "^8.2.4"
  }
}
