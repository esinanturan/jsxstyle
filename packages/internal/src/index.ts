export * as schemas from './schemas.js';
export * from './utils.js';
