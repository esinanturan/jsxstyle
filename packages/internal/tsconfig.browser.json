{
  "extends": "./tsconfig.esm.json",
  "compilerOptions": {
    "module": "ESNext",
    "moduleResolution": "bundler",
    "lib": [
      "ESNext",
      "DOM"
    ]
  }
}
