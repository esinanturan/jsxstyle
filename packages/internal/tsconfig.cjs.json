{
  "extends": "./tsconfig.base.json",
  "compilerOptions": {
    "target": "es5",
    "module": "CommonJS",
    "moduleResolution": "node10"
  }
}
