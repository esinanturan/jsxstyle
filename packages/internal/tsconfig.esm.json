{
  "extends": "./tsconfig.base.json",
  "compilerOptions": {
    "target": "ESNext",
    "module": "Node16",
    "moduleResolution": "node16",
    "lib": [
      "ESNext"
    ]
  }
}
