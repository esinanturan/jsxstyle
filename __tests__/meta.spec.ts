import { getPackList, getWorkspaces } from '@jsxstyle/internal';
import * as path from 'node:path';
import { glob, $ } from 'zx';

describe('npm publish', async () => {
  const packages = await getWorkspaces().then((pkgs) =>
    pkgs.filter((pkg) => !pkg.private)
  );

  it.for(packages)('$name only publishes the intended files', async (pkg) => {
    const packlist = await getPackList(pkg);
    expect(`
${packlist.name}
${packlist.name.replace(/./g, '=')}
${packlist.files
  .map(({ path }) => {
    // chunk filenames contain a hash that will change across builds
    const updatedFileName = path.replace(
      /(\.)[a-f0-9]{8}(\.c?js(?:\.map)?)$/,
      '$1[hash]$2'
    );
    return `- ${updatedFileName}`;
  })
  .sort()
  .join('\n')}
`).toMatchSnapshot();
  });
});

describe.skip('examples', async () => {
  const exampleDir = path.resolve(__dirname, '..', 'examples');
  const examples = await glob('jsxstyle-*-example', { cwd: exampleDir });

  for (const example of examples) {
    it(`\`${example}\` builds correctly`, async () => {
      const cwd = path.join(exampleDir, example);
      expect.assertions(1);
      return expect(() => $({ cwd })`npm run build`).not.toThrow();
    }, 30000);
  }
});
