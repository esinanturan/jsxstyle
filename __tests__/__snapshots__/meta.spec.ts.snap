// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`npm publish only publishes the intended files 1`] = `
[
  "
gatsby-plugin-jsxstyle
======================
- LICENSE
- gatsby-node.js
- gatsby-ssr.js
- index.js
- package.json
",
  "
jsxstyle
========
- LICENSE
- README.md
- lib/base64-loader/index.cjs.js
- lib/base64-loader/index.cjs.js.map
- lib/base64-loader/index.es.js
- lib/base64-loader/index.es.js.map
- lib/chunks/createClassNameGetter.[hash].cjs.js
- lib/chunks/createClassNameGetter.[hash].cjs.js.map
- lib/chunks/createClassNameGetter.[hash].es.js
- lib/chunks/createClassNameGetter.[hash].es.js.map
- lib/chunks/getStyleCache.[hash].cjs.js
- lib/chunks/getStyleCache.[hash].cjs.js.map
- lib/chunks/getStyleCache.[hash].es.js
- lib/chunks/getStyleCache.[hash].es.js.map
- lib/chunks/makeCustomProperties.[hash].cjs.js
- lib/chunks/makeCustomProperties.[hash].cjs.js.map
- lib/chunks/makeCustomProperties.[hash].es.js
- lib/chunks/makeCustomProperties.[hash].es.js.map
- lib/chunks/processProps.[hash].cjs.js
- lib/chunks/processProps.[hash].cjs.js.map
- lib/chunks/processProps.[hash].es.js
- lib/chunks/processProps.[hash].es.js.map
- lib/chunks/stringHash.[hash].cjs.js
- lib/chunks/stringHash.[hash].cjs.js.map
- lib/chunks/stringHash.[hash].es.js
- lib/chunks/stringHash.[hash].es.js.map
- lib/extract-styles/index.cjs.js
- lib/extract-styles/index.cjs.js.map
- lib/extract-styles/index.es.js
- lib/extract-styles/index.es.js.map
- lib/nextjs-plugin/index.cjs.js
- lib/nextjs-plugin/index.cjs.js.map
- lib/nextjs-plugin/index.es.js
- lib/nextjs-plugin/index.es.js.map
- lib/noop/index.cjs.js
- lib/noop/index.cjs.js.map
- lib/noop/index.es.js
- lib/noop/index.es.js.map
- lib/preact/index.cjs.js
- lib/preact/index.cjs.js.map
- lib/preact/index.es.js
- lib/preact/index.es.js.map
- lib/react/index.cjs.js
- lib/react/index.cjs.js.map
- lib/react/index.es.js
- lib/react/index.es.js.map
- lib/solid/index.cjs.js
- lib/solid/index.cjs.js.map
- lib/solid/index.es.js
- lib/solid/index.es.js.map
- lib/utils/index.cjs.js
- lib/utils/index.cjs.js.map
- lib/utils/index.es.js
- lib/utils/index.es.js.map
- lib/webpack-loader/index.cjs.js
- lib/webpack-loader/index.cjs.js.map
- lib/webpack-loader/index.es.js
- lib/webpack-loader/index.es.js.map
- lib/webpack-plugin/index.cjs.js
- lib/webpack-plugin/index.cjs.js.map
- lib/webpack-plugin/index.es.js
- lib/webpack-plugin/index.es.js.map
- package.json
- preact/lib/index.d.ts
- preact/lib/index.d.ts.map
- react/lib/componentFactory.d.ts
- react/lib/componentFactory.d.ts.map
- react/lib/index.d.ts
- react/lib/index.d.ts.map
- react/lib/styleCache.d.ts
- react/lib/styleCache.d.ts.map
- react/lib/types.d.ts
- react/lib/types.d.ts.map
- react/lib/useMatchMedia.d.ts
- react/lib/useMatchMedia.d.ts.map
- solid/lib/componentFactory.d.ts
- solid/lib/componentFactory.d.ts.map
- solid/lib/index.d.ts
- solid/lib/index.d.ts.map
- solid/lib/styleCache.d.ts
- solid/lib/styleCache.d.ts.map
- solid/lib/types.d.ts
- solid/lib/types.d.ts.map
- utils/lib/addStyleToHead.d.ts
- utils/lib/addStyleToHead.d.ts.map
- utils/lib/componentStyles.d.ts
- utils/lib/componentStyles.d.ts.map
- utils/lib/createClassNameGetter.d.ts
- utils/lib/createClassNameGetter.d.ts.map
- utils/lib/dangerousStyleValue.d.ts
- utils/lib/dangerousStyleValue.d.ts.map
- utils/lib/generateCustomPropertiesFromVariants.d.ts
- utils/lib/generateCustomPropertiesFromVariants.d.ts.map
- utils/lib/getStringHash.d.ts
- utils/lib/getStringHash.d.ts.map
- utils/lib/getStyleCache.d.ts
- utils/lib/getStyleCache.d.ts.map
- utils/lib/hyphenateStyleName.d.ts
- utils/lib/hyphenateStyleName.d.ts.map
- utils/lib/index.d.ts
- utils/lib/index.d.ts.map
- utils/lib/makeCustomProperties.d.ts
- utils/lib/makeCustomProperties.d.ts.map
- utils/lib/parseStyleProps.d.ts
- utils/lib/parseStyleProps.d.ts.map
- utils/lib/processProps.d.ts
- utils/lib/processProps.d.ts.map
- utils/lib/stringHash.d.ts
- utils/lib/stringHash.d.ts.map
- utils/lib/types.d.ts
- utils/lib/types.d.ts.map
- webpack-plugin/lib/base64Loader.d.ts
- webpack-plugin/lib/base64Loader.d.ts.map
- webpack-plugin/lib/loader.d.ts
- webpack-plugin/lib/loader.d.ts.map
- webpack-plugin/lib/nextjs.d.ts
- webpack-plugin/lib/nextjs.d.ts.map
- webpack-plugin/lib/noop.d.ts
- webpack-plugin/lib/noop.d.ts.map
- webpack-plugin/lib/plugin.d.ts
- webpack-plugin/lib/plugin.d.ts.map
- webpack-plugin/lib/types.d.ts
- webpack-plugin/lib/types.d.ts.map
- webpack-plugin/lib/utils/ModuleCache.d.ts
- webpack-plugin/lib/utils/ModuleCache.d.ts.map
- webpack-plugin/lib/utils/ast/accessSafe.d.ts
- webpack-plugin/lib/utils/ast/accessSafe.d.ts.map
- webpack-plugin/lib/utils/ast/evaluateAstNode.d.ts
- webpack-plugin/lib/utils/ast/evaluateAstNode.d.ts.map
- webpack-plugin/lib/utils/ast/extractStaticTernaries.d.ts
- webpack-plugin/lib/utils/ast/extractStaticTernaries.d.ts.map
- webpack-plugin/lib/utils/ast/extractStyles.d.ts
- webpack-plugin/lib/utils/ast/extractStyles.d.ts.map
- webpack-plugin/lib/utils/ast/generatedUid.d.ts
- webpack-plugin/lib/utils/ast/generatedUid.d.ts.map
- webpack-plugin/lib/utils/ast/getCommentForNode.d.ts
- webpack-plugin/lib/utils/ast/getCommentForNode.d.ts.map
- webpack-plugin/lib/utils/ast/getImportForSource.d.ts
- webpack-plugin/lib/utils/ast/getImportForSource.d.ts.map
- webpack-plugin/lib/utils/ast/getInlineImportString.d.ts
- webpack-plugin/lib/utils/ast/getInlineImportString.d.ts.map
- webpack-plugin/lib/utils/ast/getPropValueFromAttributes.d.ts
- webpack-plugin/lib/utils/ast/getPropValueFromAttributes.d.ts.map
- webpack-plugin/lib/utils/ast/getSourceModule.d.ts
- webpack-plugin/lib/utils/ast/getSourceModule.d.ts.map
- webpack-plugin/lib/utils/ast/getStaticBindingsForScope.d.ts
- webpack-plugin/lib/utils/ast/getStaticBindingsForScope.d.ts.map
- webpack-plugin/lib/utils/ast/parse.d.ts
- webpack-plugin/lib/utils/ast/parse.d.ts.map
- webpack-plugin/lib/utils/createModule.d.ts
- webpack-plugin/lib/utils/createModule.d.ts.map
- webpack-plugin/lib/utils/getExportsFromModuleSource.d.ts
- webpack-plugin/lib/utils/getExportsFromModuleSource.d.ts.map
- webpack-plugin/lib/utils/makePromise.d.ts
- webpack-plugin/lib/utils/makePromise.d.ts.map
- webpack-plugin/lib/utils/typePredicates.d.ts
- webpack-plugin/lib/utils/typePredicates.d.ts.map
- webpack-plugin/lib/utils/wrapFileSystem.d.ts
- webpack-plugin/lib/utils/wrapFileSystem.d.ts.map
",
]
`;
